### ISSUE TITLE
Extraneous messages when using recent version of VM

### ISSUE THREAD

**Issue by ivan-w** [SDL-Hercules-390/hyperion#293]
When running  a recent version of VM, when SRM polarization is set to vertical, the hypervisor has a tendency to put idle CPUs in a disabled wait state, and then restart them using a SIGP. Fine by me but it generates and floods the console with `HHC00814I` and `HHC00809I` messages.

It's no big deal, but it is a bit.. annoying - and we tend to loose track of actual important error messages.

Maybe a specific 'OSTAILOR' could help ? - just asking - food for thoughts.

**Comment by Peter-J-Jansen** [comment]
I too think that it should be possible to suppress these messages as they clutter the log. Either via OSTAILOR or using the MSGLEVEL command; they are probably only useful for developers debugging stuff. (I typically suppress them using a CP SET SRM POLAR HORIZ command.)

Cheers, 

Peter

**Comment by ivan-w** [comment]
Especially knowing SRM polarization doesn't make much since hercules isn't really running PR/SM !

**Comment by Fish-Git** [comment]
> I too think that it should be possible to suppress these messages as they clutter the log. Either via OSTAILOR or using the MSGLEVEL command; they are probably only useful for developers debugging stuff.

Agree 100%. The only question is _HOW_ and the heck are we going to do it?! Both messages are completely normal in virtually all other situations (that is to say, in most all situations _except_ for the one described, the messages should normally _not_ be suppressed), so how the heck are we supposed to be able to determine when they _should_ be issued and when they _shouldn't?_

Any ideas? &nbsp; `:(`

**Comment by ivan-w** [comment]
Suggestion : when OSTAILOR VM is in effect : 
- a SIGP which does not make an architecture change could be muted
- a CPU entering a disabled wait state could be muted *UNLESS* it is he last CPU in the configuration.

**Comment by Peter-J-Jansen** [comment]
Looks fine to me !

Cheers,

Peter

**Comment by Fish-Git** [comment]
I've whipped up a patch that I'm _hoping_ will resolve this issue, but before I test it, is there a fast and simple _**reliable**_ way to force the problem to occur? What do I need to do? IPL z/OS under z/VM with multiple virtual CPUs defined for it and then wait for it to go idle? Or is there an easier/faster way? Or do you just want me to just post my _(untested!)_ patch for _you_ to test for me?

**Comment by Peter-J-Jansen** [comment]
All one needs to do is IPL a z/VM >= 6.3 system with a sufficient number of CPU's.  In less than 30 seconds messages HHC00809I + HHC00814I will start appearing when the system is sufficiently idle,, like in this sample log at 18:09:46 and 18:09:50 near the bottom : 

```
18:07:18 HHC01413I Hercules version 4.3.9999.0-SDL (4.3.9999.0)
18:07:18 HHC01414I (C) Copyright 1999-2020 by Roger Bowler, Jan Jaeger, and others
18:07:18 HHC01417I CTCE v2 Fix for VM/SP + CTCE_Recovery (Peter Jansen)
18:07:18 HHC01415I Build date: Mar 31 2020 at 07:03:36
18:07:18 HHC01417I Built with: Microsoft Visual C (MSVC 192528611 0)
18:07:18 HHC01417I Build type: Windows MSVC AMD64 host architecture build
18:07:18 HHC01417I Modes: S/370 ESA/390 z/Arch
18:07:18 HHC01417I Max CPU Engines: 16
18:07:18 HHC01417I Using   shared libraries
18:07:18 HHC01417I Using   Fish threads Threading Model
18:07:18 HHC01417I Using   Error-Checking Mutex Locking Model
18:07:18 HHC01417I With    Shared Devices support
18:07:18 HHC01417I With    Dynamic loading support
18:07:18 HHC01417I With    External GUI support
18:07:18 HHC01417I With    Partial TCP keepalive support
18:07:18 HHC01417I With    IPV6 support
18:07:18 HHC01417I With    HTTP Server support
18:07:18 HHC01417I With    sqrtl support
18:07:18 HHC01417I With    Signal handling
18:07:18 HHC01417I With    Watchdog monitoring
18:07:18 HHC01417I With    CCKD BZIP2 support
18:07:18 HHC01417I With    HET BZIP2 support
18:07:18 HHC01417I With    ZLIB support
18:07:18 HHC01417I With    Regular Expressions support
18:07:18 HHC01417I With    Object REXX support
18:07:18 HHC01417I Without Regina REXX support
18:07:18 HHC01417I With    Automatic Operator support
18:07:18 HHC01417I Without National Language Support
18:07:18 HHC01417I With    CCKD64 Support
18:07:18 HHC01417I With    Transactional-Execution Facility support
18:07:18 HHC01417I Machine dependent assists: cmpxchg1 cmpxchg4 cmpxchg8 cmpxchg16 hatomics=msvcIntrinsics
18:07:18 HHC01417I Running on: PJJZ60 (Windows-6.2.9200 Intel(R) x64) LP=8, Cores=4, CPUs=1
18:07:18 HHC01417I Built with crypto external package version 1.0.0.33-g02174c9
18:07:18 HHC01417I Built with decNumber external package version 3.68.0.86-g963bd1f
18:07:18 HHC01417I Built with SoftFloat external package version 3.5.0.89-g4bd642a
18:07:18 HHC01417I Built with telnet external package version 1.0.0.50-g650fad2
18:07:18 HHC00018I Hercules is running in elevated mode
18:07:18 HHC02323W This build of Hercules has only partial TCP keepalive support
18:07:18 HHC00007I Previous message from function 'impl' at impl.c(1022)
18:07:18 HHC00150I Crypto module loaded (C) Copyright 2003-2016 by Bernard van der Helm
18:07:18 HHC00151I Activated facility: Message Security Assist
18:07:18 HHC00151I Activated facility: Message Security Assist Extension 1, 2, 3 and 4
18:07:18 HHC17528I REXX(OORexx) VERSION: REXX-ooRexx_4.2.0(MT)_64-bit 6.04 22 Feb 2014
18:07:18 HHC17529I REXX(OORexx) SOURCE:  WindowsNT
18:07:18 HHC17525I REXX(OORexx) Rexx has been started/enabled
18:07:18 HHC17500I REXX(OORexx) Mode            : Command
18:07:18 HHC17500I REXX(OORexx) MsgLevel        : Off
18:07:18 HHC17500I REXX(OORexx) MsgPrefix       : Off
18:07:18 HHC17500I REXX(OORexx) ErrPrefix       : Off
18:07:18 HHC17500I REXX(OORexx) Resolver        : On
18:07:18 HHC17500I REXX(OORexx) SysPath    (32) : On
18:07:18 HHC17500I REXX(OORexx) RexxPath   ( 0) :
18:07:18 HHC17500I REXX(OORexx) Extensions ( 8) : .REXX;.rexx;.REX;.rex;.CMD;.cmd;.RX;.rx
18:07:18 HHC00111I Thread CPU Time IS available (_POSIX_THREAD_CPUTIME=1)
18:07:18 HHC00100I Thread id 0001d640, prio 2, name 'Processor CP00' started
18:07:18 HHC00811I Processor CP00: architecture mode z/Arch
18:07:18 HHC00100I Thread id 0001a064, prio 7, name 'timer_thread' started
18:07:18 HHC01435I Config file hercules_T0.cnf: will ignore include errors
18:07:18 HHC01437I Config file[17] hercules_T0.cnf: including file hercules_host_dependent_settings.cnf
18:07:18 HHC01437I Config file[18] hercules_T0.cnf: including file hercules_4.03.cnf
18:07:18 HHC02204I ARCHLVL        set to z/Arch
18:07:18 HHC00898I Facility( 037_FP_EXTENSION ) Enabled for z/Arch
18:07:18 HHC00898I Facility( 044_PFPO ) Enabled for z/Arch
18:07:18 HHC00898I Facility( 045_DISTINCT_OPERANDS ) Enabled for z/Arch
18:07:18 HHC02204I SHCMDOPT       set to ENABLE  NODIAG8
18:07:18 HHC00898I Facility( 048_DFP_ZONE_CONV ) Enabled for z/Arch
18:07:18 HHC00898I Facility( 049_EXECUTION_HINT ) Enabled for z/Arch
18:07:18 HHC00346I cckd opts: comp=-1,compparm=-1,debug=0,freepend=-1,fsync=0,gcint=5,gcparm=0
18:07:18 HHC00346I            linuxnull=0,nosfd=0,nostress=0,ra=2,raq=4,rat=2,trace=0,wr=2
18:07:18 HHC02323W This build of Hercules has only partial TCP keepalive support
18:07:18 HHC00007I Previous message from function 'conkpalv_cmd' at hsccmd.c(7906)
18:07:18 HHC02320W Not all TCP keepalive settings honored
18:07:18 HHC00007I Previous message from function 'conkpalv_cmd' at hsccmd.c(7957)
18:07:18 HHC02204I conkpalv       set to (3,1,10)
18:07:18 HHC02204I CPUMODEL       set to 2066
18:07:18 HHC02204I CPUSERIAL      set to 012345
18:07:18 HHC02204I DIAG8CMD       set to ENABLE  ECHO
18:07:18 HHC02204I ECPSVM         set to disabled
18:07:18 HHC02204I LPARNAME       set to PJJZOS20
18:07:18 HHC02204I LPARNUM        set to 06
18:07:18 HHC17003I MAIN     storage is 12G (mainsize); storage is not locked
18:07:18 HHC02204I MOUNTED_TAPE_REINIT set to disabled
18:07:18 HHC02204I MAXCPU         set to 8
[...]
18:07:18 HHC02204I NUMCPU         set to 6
[...]
18:07:42 HHC04100I TunTap64.dll version 3.7.0.5194 initiated
[...]
18:09:10 HHC01603I ipl 200
18:09:10 HHC00811I Processor CP00: architecture mode ESA/390
18:09:10 HHC00811I Processor CP01: architecture mode ESA/390
18:09:10 HHC00811I Processor CP03: architecture mode ESA/390
18:09:10 HHC00811I Processor CP02: architecture mode ESA/390
18:09:10 HHC00811I Processor CP04: architecture mode ESA/390
18:09:10 HHC00811I Processor CP05: architecture mode ESA/390
18:09:10 HHC00814I Processor CP00: SIGP Set architecture mode            (12) CP00, PARM 00000001: CC 0
18:09:10 HHC00811I Processor CP00: architecture mode z/Arch
18:09:10 HHC00107I Starting thread cckd_ra(), active=0, started=0, max=2
18:09:10 HHC00100I Thread id 0001c5e4, prio 4, name 'cckd_ra thread 1' started
18:09:10 HHC00107I Starting thread cckd_ra() from cckd_ra(), active=1, started=1, max=2
18:09:10 HHC00100I Thread id 0001d720, prio 4, name 'cckd_ra thread 2' started
18:09:10 HHC00814I Processor CP00: SIGP Set architecture mode            (12) CP00, PARM 0000000000000000: CC 0
18:09:10 HHC00811I Processor CP00: architecture mode ESA/390
18:09:10 HHC00814I Processor CP00: SIGP Set architecture mode            (12) CP00, PARM 00000001: CC 0
18:09:10 HHC00811I Processor CP00: architecture mode z/Arch
18:09:10 HHC00814I Processor CP00: SIGP Set architecture mode            (12) CP00, PARM 0000000000000000: CC 0
18:09:10 HHC00811I Processor CP00: architecture mode ESA/390
18:09:10 HHC00814I Processor CP00: SIGP Set architecture mode            (12) CP00, PARM 00000001: CC 0
18:09:10 HHC00811I Processor CP00: architecture mode z/Arch
18:09:17 HHC00107I Starting thread cckd_writer(), active=0, started=0, max=2
18:09:17 HHC00100I Thread id 0001c2a8, prio 1, name 'cckd_writer thread 1' started
18:09:18 HHC00811I Processor CP05: architecture mode z/Arch
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP05, PARM 0000000000000000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP05, PARM 0000000000B22000: CC 0
18:09:18 HHC00811I Processor CP04: architecture mode z/Arch
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP04, PARM 0000000000000000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP04, PARM 0000000000B20000: CC 0
18:09:18 HHC00811I Processor CP03: architecture mode z/Arch
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP03, PARM 0000000000000000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP03, PARM 0000000000B1E000: CC 0
18:09:18 HHC00811I Processor CP02: architecture mode z/Arch
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP02, PARM 0000000000000000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP02, PARM 0000000000B1C000: CC 0
18:09:18 HHC00811I Processor CP01: architecture mode z/Arch
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP01, PARM 0000000000000000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP01, PARM 0000000000B1A000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP01, PARM 000000000010C658: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP01, PARM 0000000000B1A000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Restart                          (06) CP01, PARM 000000000017CDF0: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP02, PARM 000000000010C658: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP02, PARM 0000000000B1C000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Restart                          (06) CP02, PARM 000000000017CDF0: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP03, PARM 000000000010C658: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP03, PARM 0000000000B1E000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Restart                          (06) CP03, PARM 000000000017CDF0: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP04, PARM 000000000010C658: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP04, PARM 0000000000B20000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Restart                          (06) CP04, PARM 000000000017CDF0: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Initial CPU reset                (0B) CP05, PARM 000000000010C658: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Set prefix                       (0D) CP05, PARM 0000000000B22000: CC 0
18:09:18 HHC00814I Processor CP00: SIGP Restart                          (06) CP05, PARM 000000000017CDF0: CC 0
[...]
18:09:20 HHC00107I Starting thread cckd_writer() from cckd_writer(), active=1, started=1, max=2
18:09:20 HHC00107I Starting thread cckd_gcol(), active=0, started=0, max=1
18:09:20 HHC00100I Thread id 000193e8, prio 1, name 'cckd_writer thread 2' started
18:09:20 HHC00100I Thread id 0001dae4, prio 4, name 'cckd_gcol' started
18:09:46 HHC00809I Processor CP05: disabled wait state 0102400080000000 0000000000000000
18:09:46 HHC00809I Processor CP04: disabled wait state 0102400080000000 0000000000000000
18:09:50 HHC00814I Processor CP00: SIGP Restart                          (06) CP04, PARM 0000000000253C40: CC 0
18:09:50 HHC00814I Processor CP00: SIGP Restart                          (06) CP05, PARM 0000000000253C40: CC 0
[...]
18:13:38 HHC00809I Processor CP01: disabled wait state 0002000000000000 0000000000009003
18:13:38 HHC00814I Processor CP00: SIGP CPU reset                        (0C) CP01, PARM 00000000004E49A8: CC 0
18:13:38 HHC00809I Processor CP02: disabled wait state 0002000000000000 0000000000009003
18:13:38 HHC00814I Processor CP00: SIGP CPU reset                        (0C) CP02, PARM 00000000004E49A8: CC 0
18:13:38 HHC00809I Processor CP03: disabled wait state 0002000000000000 0000000000009003
18:13:38 HHC00814I Processor CP00: SIGP CPU reset                        (0C) CP03, PARM 00000000004E49A8: CC 0
18:13:38 HHC00809I Processor CP04: disabled wait state 0002000000000000 0000000000009003
18:13:38 HHC00814I Processor CP00: SIGP CPU reset                        (0C) CP04, PARM 00000000004E49A8: CC 0
18:13:38 HHC00809I Processor CP05: disabled wait state 0002000000000000 0000000000009003
18:13:38 HHC00814I Processor CP00: SIGP CPU reset                        (0C) CP05, PARM 00000000004E49A8: CC 0
```

Of course that is under the proviso that the system is still running with it's default SET SRM POLARIZATION VERTICAL. If you'd enter SET SRM POLARIZATION HORIZONTAL then the idle CPU's will no longer enter the disabled state when sufficiently idle, and be "SIGP Restart"-ed when there is again enough work to do.

Please note that the HHC00809I's at the very bottom at 18:13:38 are the result of a SHUTDOWN and these should not necessarily be OSTAILOR-d off.

Cheers,

Peter

**Comment by Fish-Git** [comment]
Thanks, Peter.

Fix by commit 97ef4d5bced55eee9d72658f87c4de159a85371d.

Closing.

(I hope I did it right! Please let me know if I didn't! It seems to work well for me.)
